docs/_blog/first_bdk_taproot_tx.md

```diff
@@ -0,0 +1,665 @@
+
+```rust
+/// Access the internal script map
+pub fn as_script_map(&self) -> &BTreeMap<(Script, LeafVersion), BTreeSet<TaprootMerkleBranch>> {}
+```
+
+Then the code would look for the "shortest" path to that specific script, as it would save size in the final transaction (leaves that are more "deep" in the tree than others naturally have more hidden branches
+in their path to the root, and thus require a longer control block to reveal them all).
+
+The issue here is that the `control_block` variable is then serialized directly into the witness. But this is not a control block, it's just a set of merkle paths! A control block only has *one* merkle path, and
+incldues the leaf version and the key parity bit.
+
+Conveniently, the `TaprootSpendInfo` struct also has this other method (I'm including the implementation as well, because it shows that internally it does the same "trick" to find the shortest path):
+
+```rust
+/// Obtain a [`ControlBlock`] for particular script with the given version.
+/// Returns [`None`] if the script is not contained in the [`TaprootSpendInfo`]
+/// If there are multiple ControlBlocks possible, this returns the shortest one.
+pub fn control_block(&self, script_ver: &(Script, LeafVersion)) -> Option<ControlBlock> {
+    let merkle_branch_set = self.script_map.get(script_ver)?;
+    // Choose the smallest one amongst the multiple script maps
+    let smallest = merkle_branch_set
+        .iter()
+        .min_by(|x, y| x.0.len().cmp(&y.0.len()))
+        .expect("Non-empty iterator");
+    Some(ControlBlock {
+        internal_key: self.internal_key,
+        output_key_parity: self.output_key_parity,
+        leaf_version: LeafVersion::default(),
+        merkle_branch: smallest.clone(),
+    })
+}
+```
+
+So to fix this code we just have to use that method instead, and we can get it done in one single line!
+
+Instead of removing the old code at the time I only commented it out, because I initially thought I would still have to look for the shortest script myself, and I figured the "sorting" code would come in handy
+later on.
+
+Also, if you are an acute observer, you might have noticed that there's a bug in this last snippet of code. Feel free to think about it a little bit, then check out the [PR][fix-control-block-bug-pr] I made
+if you wanna know the answer!
+
+```
+git show 35378ad
+```
+
+```diff
+commit 35378ad01a6f2b8161a3f36448b24d031f8aeaec
+Author: Alekos Filini <alekos.filini@gmail.com>
+Date:   Sat Nov 13 14:27:14 2021 +0100
+
+    Consider key-spend max satisfaction weight
+
+diff --git a/src/descriptor/tr.rs b/src/descriptor/tr.rs
+index 8487d56..fabf860 100644
+--- a/src/descriptor/tr.rs
++++ b/src/descriptor/tr.rs
+@@ -593,7 +593,7 @@ impl<Pk: MiniscriptKey> DescriptorTrait<Pk> for Tr<Pk> {
+     }
+
+     fn max_satisfaction_weight(&self) -> Result<usize, Error> {
+-        let mut max_wieght = None;
++        let mut max_wieght = Some(65);
```

Comment:
Note for your final PR.. the correct spelling is `weight`... the rule "i before e except after c" doesn't apply.. the English language hates it's users. 
```suggestion
+        let mut max_weight = Some(65);
```

Reply:
Yeah it was in the original code I was working on. I think I'll leave it like that in this article so that it matches the real code perfectly, but I will point it out during review on rust-miniscript